application/microservices/entando-hub-catalog-ms/src/main/java/com/entando/hub/catalog/rest/CatalogController.java

```diff
@@ -26,19 +35,30 @@ public class CatalogController {
     private final Logger logger = LoggerFactory.getLogger(CatalogController.class);
 
     private final CatalogService catalogService;
-    public CatalogController(CatalogService catalogService) {
+    private final KeycloakHelper keycloakHelper;
+
+    public CatalogController(CatalogService catalogService, KeycloakHelper keycloakHelper) {
         this.catalogService = catalogService;
+        this.keycloakHelper = keycloakHelper;
     }
 
     @Operation(summary = "Get all the catalogs", description = "Protected api, only eh-admin can access it.")
-    @RolesAllowed({ADMIN})
+    @RolesAllowed({ADMIN, AUTHOR, MANAGER})
     @ApiResponse(responseCode = "403", description = "Forbidden", content = @Content)
     @ApiResponse(responseCode = "401", description = "Unauthorized", content = @Content)
     @ApiResponse(responseCode = "200", description = "OK")
     @GetMapping(value = "/", produces = {"application/json"})
-    public ResponseEntity<List<CatalogDTO>> getCatalogs() {
+    public ResponseEntity<List<CatalogDTO>> getCatalogs(Principal principal) {
         logger.debug("REST request to get Catalogs");
-        List<CatalogDTO> catalogsDTO = catalogService.getCatalogs().stream().map(this::mapToDTO).collect(Collectors.toList());
+
+        boolean userIsAdmin = this.keycloakHelper.isAdmin(principal);
+        String username = this.keycloakHelper.getUsernameByKcToken(principal);
```

Comment:
Hi @firegloves  I'm adding a similar method in my next PR, but using the SecurityContextHolder in the SecurityHelperService,instead of passing the Principal object as parameter. If we merge my PR before this PR, probably we can simplify a bit this part of the code.

Reply:
Same about isAdmin method, we have some other similar methods in the SecurityHelperService